cpio/test/test_option_c.c

```diff
@@ -52,6 +52,9 @@ DEFINE_TEST(test_option_c)
 	char *p, *e;
 	size_t s;
 
+	/* Sanity check for casts to (int). */
+	assert(sizeof(int) >= 4);
```

Comment:
What's the desired portability for libarchive?

In order to safely parse 6 octal digits, this file requires ints that are at least 32 bits (that's a requirement from existing code, not a new thing in this commit).  If we assume that we're on a POSIX system, then we're fine.  But C99 allows for 16-bit ints.

I saw that there was some discussion about making libarchive C89-compatible, which is definitely not POSIX.  (POSIX-2008 requires C99; POSIX-2024 requires C17)   So if somebody was trying to use libarchive on an embedded system with minimal requirements, they might have 16-bit ints.

I don't mind what the answer is -- if we can assume POSIX, I'm happy to remove that sanity check.  If you want to keep the door open to other systems, I can add sanity checks where appropriate.  I could also change those sanity checks to compile-time asserts (in a C99-compatible fashion, not the C11 `_Static_assert`)

Reply:
I don't think any member of the core team cares about 16bit systems or more general, anything below 32bit.

Reply:
Getting libarchive to run on systems with 16-bit integers would be a major project, so I don't see any real point to this.

The discussion about C89 compatibility didn't really go anywhere.  The changes would be pretty dramatic, the risk of regression high, and so far no one has mentioned any specific system where that would actually be helpful.

Reply:
Ok.  Should I remove the `assert()`?  I see that you've approved the changes.

Reply:
Yes, please.

Reply:
Done; it'll take 10-15 minutes for the CI tests to run again.